### PR TITLE
chore: Replace isCreator by creatorId in ConversationViewEntity

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

- in `ConversationViewEntity` renamed field `isSelfCreated` into `isCreator` and fixed it (previously it was calculated incorrectly). 
- in `ConversationViewEntity` fixed the calculation of `isMember` field
- in `ConversationViewEntity` fixed `JOIN`
- added all the fields from `Conversation` BD table into `ConversationDetails` DB view, so now we can use only `ConversationViewEntity` model for all `SELECT` queries. This makes more sense: only 1 source of truth, fewer mappers, and more info in model.
- updated Mappers and DB-tests accordingly 
- added DB migration, so no need to clear the app data after updating.

## Need to release with [AR -1068 PR](https://github.com/wireapp/wire-android-reloaded/pull/1068)

### CODE REVIEW

**Comment by MohamadJaara** [comment]
the selfUser id is stored ion the metadata dao which is in the DB, and i remember @vitorhugods working on an quick way to get selfId from the DB

**Comment by vitorhugods** [comment]
> the selfUser id is stored ion the metadata dao which is in the DB, and i remember @vitorhugods working on an quick way to get selfId from the DB

Indeed. It's together in #1004.

#1004 is mostly reviewed, just missing some tests.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt [hunk 1]
```diff
@@ -142,8 +141,7 @@ sealed class ConversationDetails(open val conversation: Conversation) {
         val unreadMessagesCount: Long = 0L,
         val unreadMentionsCount: Long = 0L,
         val lastUnreadMessage: Message?,
-        val isSelfUserMember: Boolean = true,
-        val isSelfCreated: Boolean = false
+        val isSelfUserMember: Boolean = true
```
Comment (gongracr) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt:144]: Would this be really needed if we already have access to the same field within the `conversation` object?

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/1.sqm [hunk 2]
```diff
@@ -0,0 +1,92 @@
+DROP VIEW ConversatonDetails;
```
Comment (gongracr) [inline persistence/src/commonMain/db_user/com/wire/kalium/persistence/1.sqm:1]: Should we name this `1.sqm` or something more descriptive?

Comment (borichellow) [reply persistence/src/commonMain/db_user/com/wire/kalium/persistence/1.sqm:1]: `1.sqm` means "migration from verstion 1 to version 2", so it has to be `1` :)

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt (hunk 1 above)
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt:144]: that's right, this is why i removed that field here :)

Comment (gongracr) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt:144]: But `isSelfUserMember` is still part of the `Conversation` data class in line 30 🤷

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepositoryTest.kt [hunk 3]
```diff
@@ -437,6 +438,11 @@ class ConversationRepositoryTest {
             .whenInvokedWith(anything(), anything())
             .thenDoNothing()
 
+        given(conversationDAO)
+            .suspendFunction(conversationDAO::getConversationByQualifiedID)
+            .whenInvokedWith(anything())
+            .then { TestConversation.VIEW_ENTITY }
+
```
Comment (gongracr) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepositoryTest.kt:445]: I might sound like a scratched vinyl, but if you have time, adapting this test to the Arrangement pattern will save us a lot of boilerplate code and make it easier to add new one :)

Comment (borichellow) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepositoryTest.kt:444]: i was afraid you ask for that XD

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt (hunk 1 above)
Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt]: after your question i noticed that both `isSelfUserMember` and `isSelfCreated ` should be not in `Conversation` but in `ConversationDetails.OneOne` and `ConversationDetails.Group` as these fields are related only to these data classes. So I moved them :)

**Review by gongracr — APPROVED** [review]
Thanks for the changes 🙇 And glad that my comments could help you identify some other potential bugs 😄  Great job!

**Review by mchenani — APPROVED** [review]
Well done buddy 🚀 fixed many issues and cleaned up the way more than before 🥁

**Pull request merged**